streams/src/test/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImplTest.java

```diff
@@ -171,6 +171,28 @@ public void shouldReadCheckpointOffsets() throws IOException {
         assertEquals(expected, offsets);
     }
 
+    @Test(expected = StreamsException.class)
```

Comment:
It's better to not use this annotation, because if any other line than the expected one throws, the test would incorrectly pass.

It seems some existing tests, also use this undesired pattern. Feel free to do a MINOR follow up PR to improve the test code.